#### Group 7 Members: Tustin Choa, Margaret Inguillo, Ryan Villanueva

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

In [11]:
stats_df = pd.read_csv("marketing_campaign.csv", sep="\t")

In [12]:
stats_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

This dataset is about customer personality types. It contains 2240 observations and 29 variables. Those variables are grouped into 4 major categories: People, products, promotion, and place. This was created in order to help companies create products that may be sold to a certain group of people with similar traits. It was likely that the information was gathered through the company's website in the form of cookies, as well as, through membership data. 

Listed below are the variables per category and their interpretation:                           
**People**                                   
ID: Unique number assigned to every customer                               
Year_Birth: Customer's year of birth                                   
Education: Customer's education attainment                             
Marital_Status: Customer's marital status                                
Income: Customer's yearly income                                  
Kidhome: Number of children a customer has                                                       
Teenhome: How many of the children are teenagers                            
Dt_Customer: Customer's date of enrollment at the company                     
Recency: Number of days since customer's last purchase                                
Complain: 1 if a complain was made by the customer in the last 2 years , 0 otherwise                      

**Products**                                                            
MntWines: Expenses on wine in the last 2 years                 
MntFruits: Expenses on fruits in the last 2 years                                    
MntMeatProducts: Expenses on meat in the last 2 years                                   
MntFishProducts: Expenses on fish in the last 2 years                                           
MntSweetProducts: Expenses on sweets the in last 2 years                            
MntGoldProds: Expenses on gold in the last 2 years                       

**Promotion**                                     
NumDealsPurchases: The number of discounted purchases
AcceptedCmp1: 1 if customer bought an offer in the 1st campaign, 0 otherwise                    
AcceptedCmp2: 1 if customer bought an offer in the 2nd campaign, 0 otherwise                     
AcceptedCmp3: 1 if customer bought an offer in the 3rd campaign, 0 otherwise                     
AcceptedCmp4: 1 if customer bought an offer in the 4th campaign, 0 otherwise                              
AcceptedCmp5: 1 if customer bought an offer in the 5th campaign, 0 otherwise                             
Response: 1 if customer bought an offer in the 6th campaign, 0 otherwise                     

**Place**                                          
NumWebPurchases: The number of online purchases                                    
NumCatalogPurchases: The number of purchases through a catalogue                               
NumStorePurchases: The number of purchases made directly in stores                             
NumWebVisitsMonth: The number of visits to company’s web site in the last month 

#### **Still looking for these**
Z_CostContact: Amount of contact with the customer (?)                                     
Z_Revenue: 

In [21]:
stats_df.describe().T

count          mean           std     min       25%  \
ID                   2240.0   5592.159821   3246.662198     0.0   2828.25   
Year_Birth           2240.0   1968.805804     11.984069  1893.0   1959.00   
Income               2216.0  52247.251354  25173.076661  1730.0  35303.00   
Kidhome              2240.0      0.444196      0.538398     0.0      0.00   
Teenhome             2240.0      0.506250      0.544538     0.0      0.00   
Recency              2240.0     49.109375     28.962453     0.0     24.00   
MntWines             2240.0    303.935714    336.597393     0.0     23.75   
MntFruits            2240.0     26.302232     39.773434     0.0      1.00   
MntMeatProducts      2240.0    166.950000    225.715373     0.0     16.00   
MntFishProducts      2240.0     37.525446     54.628979     0.0      3.00   
MntSweetProducts     2240.0     27.062946     41.280498     0.0      1.00   
MntGoldProds         2240.0     44.021875     52.167439     0.0      9.00   
NumDealsPurchases    2240.0      2.325000      1.932238     0.0      1.00   
NumWebPurchases      2240.0      4.084821      2.778714     0.0      2.00   
NumCatalogPurchases  2240.0      2.662054      2.923101     0.0      0.00   
NumStorePurchases    2240.0      5.790179      3.250958     0.0      3.00   
NumWebVisitsMonth    2240.0      5.316518      2.426645     0.0      3.00   
AcceptedCmp3         2240.0      0.072768      0.259813     0.0      0.00   
AcceptedCmp4         2240.0      0.074554      0.262728     0.0      0.00   
AcceptedCmp5         2240.0      0.072768      0.259813     0.0      0.00   
AcceptedCmp1         2240.0      0.064286      0.245316     0.0      0.00   
AcceptedCmp2         2240.0      0.013393      0.114976     0.0      0.00   
Complain             2240.0      0.009375      0.096391     0.0      0.00   
Z_CostContact        2240.0      3.000000      0.000000     3.0      3.00   
Z_Revenue            2240.0     11.000000      0.000000    11.0     11.00   
Response             2240.0      0.149107      0.356274     0.0      0.00   

                         50%       75%       max  
ID                    5458.5   8427.75   11191.0  
Year_Birth            1970.0   1977.00    1996.0  
Income               51381.5  68522.00  666666.0  
Kidhome                  0.0      1.00       2.0  
Teenhome                 0.0      1.00       2.0  
Recency                 49.0     74.00      99.0  
MntWines               173.5    504.25    1493.0  
MntFruits                8.0     33.00     199.0  
MntMeatProducts         67.0    232.00    1725.0  
MntFishProducts         12.0     50.00     259.0  
MntSweetProducts         8.0     33.00     263.0  
MntGoldProds            24.0     56.00     362.0  
NumDealsPurchases        2.0      3.00      15.0  
NumWebPurchases          4.0      6.00      27.0  
NumCatalogPurchases      2.0      4.00      28.0  
NumStorePurchases        5.0      8.00      13.0  
NumWebVisitsMonth        6.0      7.00      20.0  
AcceptedCmp3             0.0      0.00       1.0  
AcceptedCmp4             0.0      0.00       1.0  
AcceptedCmp5             0.0      0.00       1.0  
AcceptedCmp1             0.0      0.00       1.0  
AcceptedCmp2             0.0      0.00       1.0  
Complain                 0.0      0.00       1.0  
Z_CostContact            3.0      3.00       3.0  
Z_Revenue               11.0     11.00      11.0  
Response                 0.0      0.00       1.0